pageserver/src/bin/pageserver.rs

```diff
@@ -276,7 +275,17 @@ fn start_pageserver(
     let pageserver_listener = tcp_listener::bind(pg_addr)?;
 
     // Launch broker client
-    WALRECEIVER_RUNTIME.block_on(pageserver::broker_client::init_broker_client(conf))?;
+    let broker_client = WALRECEIVER_RUNTIME
+        .block_on(async {
+            // Note: we do not attempt connecting here (but validate endpoints sanity).
+            storage_broker::connect(conf.broker_endpoint.clone(), conf.broker_keepalive_interval)
+        })
```

Comment:
`storage_broker::connect` is not async, no need for the block_on here.

Reply:
Yes there is, the storage_broker::connect / grpc tonic crate requires running inside a tokio runtime thread. I'll add a comment.

Reply:
Gotcha. I'd suggest [Runtime::entrer](https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#method.enterl) rather than block_on, then.